alf/networks/projection_networks.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import functools
+from typing import Callable
+
+import torch
+import torch.nn as nn
+import torch.distributions as td
+
+import alf.layers as layers
+from alf.tensor_specs import TensorSpec, BoundedTensorSpec
+
+
+def DiagMultivariateNormal(loc, scale_diag):
+    return td.Independent(td.Normal(loc, scale_diag), 1)
+
+
+class CategoricalProjectionNetwork(nn.Module):
```

Comment:
tf_agent (Categorical|Normal)ProjectionNetwork are subclass of DistributionNetwork, which has a property output_spec. I'm not sure we will need it or not. Just keep it in mind that we may have to implement it later.